### ISSUE TITLE
Arabic rendered as Left to Right

### ISSUE THREAD

**Issue by hubaishan** [microsoft/terminal#17904]
### Windows Terminal version

 1.22.240823002-preview

### Windows build number

10.0.19045.4529

### Steps to reproduce

write Arabic text

### Expected Behavior

![image](https://github.com/user-attachments/assets/d5b50a33-9012-4c44-84a4-45e232a3bcbc)
from ver  1.19.11213.0 disabling "AtlasEngine" 

### Actual Behavior

![image](https://github.com/user-attachments/assets/f55efdf7-6da4-4fb1-8054-5067968809ae)

**Comment by lhecker** [comment]
I did this intentionally so that the Arabic mode in (neo)vim would work (`vim -A`). For instance, using the sample text from https://r12a.github.io/scripts/arab/arb.html:

![image](https://github.com/user-attachments/assets/b405462d-fcc2-4185-ac63-8ee50fad11e6)

I was hoping to add a setting in the future which enables RTL overrides and/or detection, but I haven't gotten there yet myself.

**Comment by zadjii-msft** [comment]
Probably should just bucket that with #538, yea?

**Comment by lhecker** [comment]
Yeah, I'll close this as a /duplicate of #538. Thank you for the report though!

**Comment by hubaishan** [comment]
> Yeah, I'll close this as a /duplicate of #538. Thank you for the report though!

@lhecker these are tow different  bugs this is new bug in AtlasEngine, #538 is old bug before AtlasEngine.

**Comment by DHowett** [comment]
#538 tracks us _actually supporting RTL text_ properly. It is the right issue for RTL issues because we have not officially declared support for RTL, on the old or the new renderer 😄

**Comment by lhecker** [comment]
It's also worth mentioning again IMO, that I did this intentionally, so it's definitely not a bug. I did that because `vim -A` among others actively expects a terminal to not support RTL. I suspect it'll be easier to add RTL support once the rendering architecture has been refactored, because then AtlasEngine will be a fraction of its current size.
